src/figuresClasses.ts

```diff
@@ -1,19 +1,81 @@
-export interface Figure {
-
+export function zeroChecker(...args: number[]): boolean {
+  return args.every((number) => number >= 0);
 }
 
-export class Triangle {
+export enum FigureShapes {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
 
+export interface Figure {
+  shape: FigureShapes,
+  color: string,
+  getArea: Function,
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape = FigureShapes.Triangle;
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (!zeroChecker(a, b, c)) {
+      throw new Error('Triangle sides should be greater than 0');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('Cannot form a triangle. One side is too long bro/sis');
+    }
+  }
 
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+
+    const area = Math.sqrt(s * ((s - this.a) * (s - this.b) * (s - this.c)));
+
+    return Math.floor(area * 100) / 100;
```

Comment:
you round with the same logic twice over the code. You also could create a separate helper function - `rounder` for example for this purpose.

Reply:
Thank you for comments!